src/server/game/Combat/ThreatManager.cpp

```diff
@@ -415,11 +416,28 @@ void ThreatManager::addThreat(Unit* victim, float threat, SpellSchoolMask school
 void ThreatManager::doAddThreat(Unit* victim, float threat)
 {
     uint32 redirectThreadPct = victim->GetRedirectThreatPercent();
+    Unit* redirectTarget = victim->GetRedirectThreatTarget();
```

Comment:
Why is this weird threat redirection needed? Shouln't you make it that personal spawns cannot see any players except their "owner" (and his pets/guardians/other summons) (pretty sure this isnt here for aggro on spawn event)

Reply:
Because personal spawns can attack non-personal spawns, the attacked creature must attack the owner and not the personal spawn